admin-dev/themes/new-theme/js/pages/product/components/categories/categories-manager.ts

```diff
@@ -0,0 +1,152 @@
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+import ProductMap from '@pages/product/product-map';
+import ProductEventMap from '@pages/product/product-event-map';
+import CategoryTreeSelector from '@pages/product/components/categories/category-tree-selector';
+import Tags from '@pages/product/components/categories/tags';
+import {EventEmitter} from "events";
+
+const ProductCategoryMap = ProductMap.categories;
+
+export default class CategoriesManager {
+  eventEmitter: EventEmitter;
+
+  categoryTreeSelector: CategoryTreeSelector;
+
+  categoriesContainer: HTMLElement;
+
+  addCategoriesBtn: HTMLElement;
+
+  typeaheadData: Array<any>;
+
+  tags: Tags;
+
+  constructor(eventEmitter: EventEmitter) {
+    this.eventEmitter = eventEmitter;
+    this.categoryTreeSelector = new CategoryTreeSelector(eventEmitter);
+
+    const categoriesContainer = document.querySelector(ProductCategoryMap.categoriesContainer);
+
+    if (!(categoriesContainer instanceof HTMLElement)) {
+      throw '"categoriesContainer" must be a valid HTMLElement';
```

Comment:
I see what you mean regarding errors, and indeed throwing one here might break the rest of the JS code

That said if one core component is not working correctly we can consider the page buggy, so IMHO it's worth throwing an error the sooner we can when we detect something is wrong (invalid selector in mapping, modified templates, ...)

Reply:
But it seems like you need to throw an object there however
```suggestion
      throw new Error('"categoriesContainer" must be a valid HTMLElement');
```

Reply:
I've discussed with @NeOMakinG, he mentioned its possibly not worth breaking whole js, instead use console.error() and return where possible.